fcrepo-kernel-modeshape/src/test/java/org/fcrepo/kernel/modeshape/FedoraResourceImplTest.java

```diff
@@ -150,6 +166,48 @@ public void testGetCreatedDate() throws RepositoryException {
     }
 
     @Test
+    public void testGetTypes() throws RepositoryException {
+        final String mockNodeTypePrefix = "jcr";
+
+        final String mockPrimaryNodeTypeName = "somePrimaryType";
+        final String mockMixinNodeTypeName = "someMixinType";
+        final String mockPrimarySuperNodeTypeName = "somePrimarySuperType";
+        final String mockMixinSuperNodeTypeName = "someMixinSuperType";
+
+        final Workspace mockWorkspace = mock(Workspace.class);
+        final NamespaceRegistry mockNamespaceRegistry = mock(NamespaceRegistry.class);
+
+        when(mockSession.getWorkspace()).thenReturn(mockWorkspace);
+        when(mockWorkspace.getNamespaceRegistry()).thenReturn(mockNamespaceRegistry);
+        when(mockNamespaceRegistry.getURI("jcr")).thenReturn(JCR_NAMESPACE);
+        when(mockNode.getPrimaryNodeType()).thenReturn(mockPrimaryNodeType);
+        when(mockPrimaryNodeType.getName()).thenReturn(
+                mockNodeTypePrefix + ":" + mockPrimaryNodeTypeName);
+
+        when(mockNode.getMixinNodeTypes()).thenReturn(
+                new NodeType[]{mockMixinNodeType});
+        when(mockMixinNodeType.getName()).thenReturn(
+                mockNodeTypePrefix + ":" + mockMixinNodeTypeName);
+
+        when(mockPrimaryNodeType.getSupertypes()).thenReturn(
+                new NodeType[]{mockPrimarySuperNodeType});
+        when(mockPrimarySuperNodeType.getName()).thenReturn(
+                mockNodeTypePrefix + ":" + mockPrimarySuperNodeTypeName);
+
+        when(mockMixinNodeType.getSupertypes()).thenReturn(
+                new NodeType[] {mockMixinSuperNodeType, mockMixinSuperNodeType});
+        when(mockMixinSuperNodeType.getName()).thenReturn(
+                mockNodeTypePrefix + ":" + mockMixinSuperNodeTypeName);
+
+        final List<URI> types = testObj.getTypes();
+        assertTrue(types.contains(URI.create(REPOSITORY_NAMESPACE + "somePrimaryType")));
```

Comment:
A minor point, but you could use the constants defined above instead of recreating the Strings.